testing/test_recwarn.py

```diff
@@ -106,31 +106,69 @@ def test_deprecated_explicit_call(self):
         pytest.deprecated_call(self.dep_explicit, 0)
         pytest.deprecated_call(self.dep_explicit, 0)
 
-    def test_deprecated_call_as_context_manager_no_warning(self):
-        with pytest.raises(pytest.fail.Exception, matches='^DID NOT WARN'):
-            with pytest.deprecated_call():
-                self.dep(1)
+    @pytest.mark.parametrize('mode', ['context_manager', 'call'])
+    def test_deprecated_call_no_warning(self, mode):
+        """Ensure deprecated_call() raises the expected failure when its block/function does
+        not raise a deprecation warning.
+        """
+        def f():
+            pass
+
+        msg = 'Did not produce DeprecationWarning or PendingDeprecationWarning'
```

Comment:
Now both context-manager form and function-call form produce the same message; note that previously the context-manager version produced a `"DID NOT WARN..."` message from `pytest.warns` instead (see the removed lines just above this one).

I think the *vast* majority of users just use `pytest.deprecated_call` to ensure things are issuing deprecation warnings, not actually capturing the AssertionError and checking the message; the latter only makes sense when testing that `pytest.deprecated_call` works and raises the expected message, which I think it is done only pytest's own test suite.

Having said that, do you feel because of this we should play safe and merge this into features instead?